src/main/java/micdoodle8/mods/galacticraft/api/spaceprojects/SpaceProject.java

```diff
@@ -0,0 +1,40 @@
+package micdoodle8.mods.galacticraft.api.spaceprojects;
+
+import java.util.HashMap;
+import java.util.UUID;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.server.MinecraftServer;
+
+public class SpaceProject {
+    // Maps UUIDs to other UUIDs. This allows users to join a team.
+    public static HashMap<UUID, UUID> GlobalSpaceProjectTeam = new HashMap<>(100, 0.9f);
+
+    public static void joinUserNetwork(UUID user_uuid_0, UUID user_uuid_1) {
+        GlobalSpaceProjectTeam.put(user_uuid_0, user_uuid_1);
+    }
+
+    public static void strongCheckOrAddUser(EntityPlayer user) {
+        strongCheckOrAddUser(user.getUniqueID());
+    }
+
+    public static void strongCheckOrAddUser(UUID user_uuid) {
+
+        // Check if the user has a team. Add them if not.
+        GlobalSpaceProjectTeam.putIfAbsent(user_uuid, user_uuid);
+    }
+
+    public static String GetUsernameFromUUID(UUID uuid) {
```

Comment:
`GetUsernameFromUUID` should start with 'g'

Reply:
anything else i should fix like that?

Reply:
Not that I've seen. You should think about how you check against empty UUIDs though, using `equals("")` not only probably won't work, it will likely cause a NullPointerException, since iirc the PlayerProfileCache will return a profile with null uuid.

Reply:
i think that was used because everything was in string the the player UUID could sometimes not be in the Map, but now i don't think we can check for that as we are now using minecraft to get the UUID

Reply:
At most i think i could check if its null and send a player message saying: player not found, depending on which user it was

Reply:
but if I check for null it says that it never can be null so /shrug